Cleanup script and new docker image in order to clean-up rabbitmq and mongodb resources

val-andrushko opened the pull request:
Not sure what to explain here, it's all in the readme and this is basically the initial commit/push :)

val-andrushko committed:
Add initial files, script and readme are still WIP

val-andrushko committed:
change author

val-andrushko committed:
Adjust script, remove redundant stuff, improve readability

val-andrushko committed:
Update readme, added suggested newlines

val-andrushko committed:
update readme

pascal89 commented on Dockerfile:
```diff
@@ -0,0 +1,26 @@
+FROM alpine:edge
```
Maybe it's better to go for [latest](https://github.com/alpinelinux/docker-alpine/blob/adebcfd075c83ef788b5f071678dcfb8ea118bb3/x86_64/Dockerfile), edge may contain bugs.

val-andrushko replied:
Changed to alpine:latest

pascal89 commented on README.md:
```diff
@@ -0,0 +1,40 @@
+
+## environment variables
+
+These environment variables MUST be set! Purpose of this container image is to automatically delete resources after each feature(dev) deployment is deleted from k8s. Database/exchanges names will be different on every run.
```
This sentence doesn't read well.

val-andrushko replied:
What about 

`These environment variables are mandatory and must be set before running this container.` 

?

Mastermindzh replied:
The second one is more formal but there is nothing wrong with the grammar of the first actually :)

Mastermindzh commented on Dockerfile:
```diff
@@ -0,0 +1,26 @@
+
+WORKDIR /source
+ENV RABBITMQADMIN_VERSION="v3.9.0"
+RUN apk add --no-cache ca-certificates bash git openssh curl \
```
please lint this file with HADOlint and follow its advice :)

george-nt replied:
@Mastermindzh would be better to use `alpine:3.16.2` instead of using `alpine:latest`, one of the rules that hadolint is introducing is this one:
1 DL3007 warning: Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag

george-nt replied:
![image](https://user-images.githubusercontent.com/111972490/187414316-db884de3-f718-454f-a57e-ee4ef83e0fef.png)
These are the remaining hadolint warnings.
Do you want to try add a specific version tag for each of the dependencies?

Mastermindzh replied:
I usually don't bother myself tbh :)

val-andrushko replied:
We decided to go with the latest, it turned out to be was quite a job to get all the exact versions.

val-andrushko committed:
Use alpine:latest

Improve readme

Mastermindzh commented on README.md:
```diff
@@ -0,0 +1,40 @@
+
+# dev-deployment-cleanup docker image
+
+This container image can be used to automatically drop mongoDB databases en clean-up rabbitmq resources like an exchange, its bindings and destion queues.
```
en == and

Mastermindzh replied:
destion == destination.

And whilst you're at it "clean up" is either 1 word (merica!), or 2: "clean up" 😛

Mastermindzh commented on scripts/resources-cleanup-deployment.sh:
```diff
@@ -0,0 +1,31 @@
+
+# MONGO_URI and MONGO_DB are already set as environment variables
+# connect to mongoDb and drop database 
+mongo "${MONGO_URI}" --eval "db.getSiblingDB('${MONGO_DB}');db.dropDatabase()"
```
Can we add a check that "admin" will not be deleted ?
Removing the admin db would be problematic (that would bring EVERYTHING down as it governs login.

val-andrushko replied:
Check was added by George

Mastermindzh commented on scripts/resources-cleanup-deployment.sh:
```diff
@@ -0,0 +1,31 @@
+
+# MESSAGE_QUEUE_URI and EXCHANGE_NAME are already set as environment variables
+# Extract host, username and password from the MESSAGE_QUEUE_URI connection string
+host=$(echo $MESSAGE_QUEUE_URI | cut -d "@" -f 2 | cut -d "/" -f 1)
```
Please add `"` around **ALL** variables to prevent word splitting.

You can use "shellcheck" if you want, I'm sure it'll complain about word splitting haha!
(who knows, it might even find more! But your script looks quite clean)

PS: check out `awk`, could've made some things (like selecting columns) less error prone. (tho not necessary...)

val-andrushko replied:
Also fixed by George

Mastermindzh reviewed (changes requested):
I would like to request some changes (especially the admin thing and the docker cleanup).

val-andrushko committed:
Fix typos in README.md

HBTeun reviewed (approved):
If you do the other comments

Commit:
Linting the Dockerfile by using hadolint
Added quotes for the variable names
Added check in order to not delete the admin db

george-nt commented on scripts/resources-cleanup-deployment.sh:
```diff
@@ -0,0 +1,25 @@
+# 1. MongoDB cleanup
+# MONGO_URI and MONGO_DB are already set as environment variables
+# connect to mongoDb and drop database 
+if [[ ${MONGO_DB} != *"admin"* ]]; then
```
Added a check for not deleting the admin db.

Commit:
Fixed hadolint info message
